src/util/mogps.jl

```diff
@@ -0,0 +1,105 @@
+"""
+    MOutput(x::AbstractVector, out_dim::Integer)
+
+A data type to handle multi-dimensional outputs.
+"""
+struct MOutput{T<:AbstractVector} <: AbstractVector{Real}
+    x::T
+    out_dim::Integer
+end
+
+Base.size(out::MOutput) = (out.out_dim * size(out.x, 1),)
+
+@inline function Base.getindex(out::MOutput, ind::Integer)
+    @boundscheck checkbounds(out, ind)
+    len = length(out.x)
+    ind1 = ind % len
+    ind2 = (ind - 1) ÷ len + 1
+    if ind1 == 0 ind1 = len end
```

Comment:
If I'm not mistaken, this can be written as
```suggestion
    ind1, ind2 = fldmod1(ind, length(out.x))
```

Reply:
No they are not the same. All the tests fail if I replace it.

Reply:
Hmm I guess you should be able to use `fld1` and `mod1` instead of the current implementation but I assume I messed up the remainder etc. E.g., `ind1` is just `mod1(ind, len)`. So maybe it's just
 ```suggestion
    ind2, ind1 = fldmod1(ind, length(out.x))
```
?

Reply:
This worked. Thanks!